ckanext/cfpb_extrafields/plugin.py

```diff
@@ -35,8 +36,71 @@ def popup_usage_restrictions():
     return "Enter instructions for what users can and cannot do with the data."
 
 class ExampleIDatasetFormPlugin(p.SingletonPlugin, tk.DefaultDatasetForm):
-    p.implements(p.IDatasetForm)
-    p.implements(p.IConfigurer)
+
+    p.implements(p.IResourceController)
+    def before_create(self, context, resource):
+        return
+    def after_create(self, context, resource):
+        # must have changed['format'] when you created the resource
+        tk.redirect_to(controller='package',action='resource_edit',
+                       resource_id=resource['id'],id=resource['package_id'])
+
+    def _control_datadict(self, resource):
+        ''' delete and rebuild datadict'''
+        record = resource.get('datadict','')
+        if record:
+            resource.pop('datadict', None)
+            record = eval(record) 
+            ds.delete_datastore_json(resource['id'], 'name', 'datadict')
+            ds.create_datastore_json(resource['id'], record, 'name', 'datadict') 
+            # print 'check datadict? ',  ds.print_datastore_json(resource['id'],'','') 
+        return
+    def before_update(self, context, current, resource):
+        # note keys that have changed (resource is new current is old)
+        self.changed_keys = ['format', 'privacy_contains_pii']
+        self.changed = {}
+        for i in self.changed_keys:
+            self.changed[i] = False
+            if resource.get(i,'0') != current.get(i,'1'):
+                print 'trigger a redirect in after_update: ', self.changed.get(i,'')
+                self.changed[i] = True
+        # mimic a new controller for data dictionary field
+        if current.get('format','0') == 'Data Dictionary' and resource.get('format','1') == 'Data Dictionary':
+            self._control_datadict(resource)
+
```

Comment:
awesome! just tweaked it to dictionary comprehension because I want to know whether each element changed:
self.changed = {key: resource.get(key, '0') != current.get(key, '1') for key in ['format', 'privacy_contains_pii']}